src/app/workflow/workflow.component.ts

```diff
@@ -352,6 +353,18 @@ export class WorkflowComponent extends Entry implements AfterViewInit {
     }
   }
 
+  requestDOIForWorkflowVersion() {
+    const message = 'Creating DOI';
```

Comment:
Throw majority of the code into the service and add some null checks for this.workflow and this.selectedVersion

Reply:
https://ucsc-cgl.atlassian.net/browse/DOCK-830
For a later sprint